### ISSUE TITLE
false negative with numbers using jsx-pascal-case

### ISSUE THREAD

**Issue by remitbri** [yannickcr/eslint-plugin-react#339]
Using v3.11.1, 
```jsx
<My1ComponentBad />
<MyComponent1Bad />
<MyComponentBad1 />
```
flag an error (e.g `Imported JSX component MyComponentBad1 must be in PascalCase (react/jsx-pascal-case)`)
